test/utils_test.js

```diff
@@ -617,6 +859,120 @@ describe("utils", () => {
         });
       });
 
+      it("should populate defaults for nested dependencies in arrays", () => {
+        const schema = {
+          type: "array",
+          items: {
+            properties: {
+              foo: {
+                type: "object",
+                properties: {
+                  name: {
+                    type: "string",
+                  },
+                },
+                dependencies: {
+                  name: {
+                    oneOf: [
+                      {
+                        properties: {
+                          name: {
+                            type: "string",
+                          },
+                          grade: {
+                            type: "string",
+                            default: "A",
+                          },
+                        },
+                      },
+                    ],
+                  },
+                },
+              },
+            },
+          },
+        };
+        expect(getDefaultFormState(schema, [{ foo: { name: "Name" } }])).eql([
```

Comment:
Can you add another test in which the formData is empty, name has a default of "Name", and thus the result should have both name and grade?

Reply:
Nice catch. That actually fails (when formData is`[{}]`, which is the only time the code actually injects the defaults here). But this already fails when dependencies are not used within arrays.
I have to check if/how we can fix this. 
But I guess it does make sense that the dependencies are evaluated (again) after injecting the defaults.

Reply:
Yeah, if you can fix it now, please do that, otherwise, we can fix that in a follow-up PR.